src/components/InitialMap.js

```diff
@@ -118,45 +118,55 @@ class MapContainer extends Component {
 
```

Comment:
This was a really difficult piece of code to go through.  I'm not sure what's all happening here and it gets kind of messy, but I think this was not your code, but from one of the packages. haha

Reply:
@amr08 , the package is used, but most of the code is done by the team. We would be refactoring it to include Infowindow portions as a separate component. Ideally whatever possible can be broken into smaller components and I see the scope in InfoWindow. Marker is also another one, but we can first try segregating InfoWindow from map and see if any issues are there.